src/current/v24.1/certificate-based-authentication-using-the-x509-subject-field.md

```diff
@@ -0,0 +1,118 @@
+
+## Example
+
+This example shows how to set up a mapping between the Subject field of an [X.509 certificate](https://en.wikipedia.org/wiki/X.509) and a SQL [user or role]({% link {{page.version.version}}/security-reference/authorization.md %}#roles).
+
+### Step 1. Create certificates using your infrastructure
+
+These instructions assume that you have already created certificates using your own infrastructure for the following users:
+
+- The [`root` user]({% link {{ page.version.version }}/security-reference/authorization.md %}#root-user).
+- The [`node` user]({% link {{ page.version.version }}/security-reference/authorization.md %}#node-user).
+- Any [SQL user]({% link {{ page.version.version }}/security-reference/authorization.md %}#sql-users) who needs to authenticate with your cluster. In this example, we will call the user `maxroach`.
+
+### Step 2. Start your cluster with root and node certificate flags
+
+At cluster startup, you'll need to pass the [`cockroach start`]({% link {{ page.version.version }}/cockroach-start.md %}) flags [`--node-cert-distinguished-name`]({% link {{ page.version.version }}/cockroach-start.md %}#flags-node-cert-distinguished-name) and [`--root-cert-distinguished-name`]({% link {{ page.version.version }}/cockroach-start.md %}#flags-root-cert-distinguished-name).
+
+The argument to each flag is a string with a comma separated list of distinguished name (DN) mappings in `{attribute-type}={attribute-value}` format in accordance with [RFC4514](https://www.rfc-editor.org/rfc/rfc4514). When each of these flags are set, the argument needs to be an exact match with the DN subject in the client certificate provided.
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+cockroach start --certs-dir=/path/to/certs --node-cert-distinguished-name="O=Cockroach,CN=node" --root-cert-distinguished-name="O=Cockroach,CN=root" # other startup flags, etc.
+~~~
+
+The DN mappings in the certificates you create should match what you pass in at cluster start time. One way to create a certificate matching the `node` user in the example above is:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+openssl req -new -key /path/to/certs/client.node.key -out client.node.csr -batch -subj /O=Cockroach/CN=node
+~~~
+
+The same command can be used to generate a certificate for the `root` user.
+
+### Step 3. Enable the cluster setting to require the Subject field in certificates
+
+Once the cluster is started, enable the following [cluster setting]({% link {{ page.version.version }}/cluster-settings.md %}#setting-security-client-cert-subject-required-enabled):
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SET CLUSTER SETTING security.client_cert.subject_required.enabled = true;
+~~~
+
+When the setting is enabled, CockroachDB will verify the following during user authentication: 
```

Comment:
These verifications are actually done regardless of the cluster setting, as long as the subject/DN options are set. What happens when the `subject_required` setting is enabled is that users who don't have a SUBJECT role option set will be unable to log in with certificates.

Reply:
I've modified this to say

> When this setting is enabled, users who don't have a `SUBJECT` role option set will be unable to log in with certificates.
>
> Regardless of the value of this setting, CockroachDB will verify the following during user authentication as long as the Subject/DN options are set:
> - ....